### PR TITLE
Use ES6 modules, use webpack bundler (cont.)

### PR DESCRIPTION
**Pull request by tyrasd**
continuation of #282 (moving onto a branch on the main github repo for easier collaboration @simon04).

* [x] install packages from npm via
* [x] use ES6 module `export`/`import`
* [x] use [webpack](https://webpack.js.org/) for bundling the ide (`index.html`)
* [x] use [webpack](https://webpack.js.org/) for bundling the interactive map (`map.html`)
* [x] make the tests run again
* [x] get rid of `*_packaged.html` files
* [x] switch to different publishing mechanism: use gh-pages branch to contain only bundled stuff (like [this](https://github.com/hotosm/osm-analytics/blob/master/deploy.sh) maybe)?

### CODE REVIEW

File: js/ide.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ import CodeMirror from 'codemirror/lib/codemirror.js';
 import moment from 'moment';
 import tokml from 'tokml';
 import togpx from 'togpx';
-import * as _saveAs from 'file-saver';
+import {saveAs} from 'file-saver';
```
Comment (tyrasd) [inline js/ide.js]: thanks. I couldn't find out how to get this to run properly on the weekend. I still have to get used to the new import syntax :man_shrugging:

Comment (simon04) [inline js/ide.js]: The best introduction into ES6 modules I've found is [the chapter in Exploring ES6](http://exploringjs.com/es6/ch_modules.html).

**Comment by simon04** [comment]
Jay, all tests finally succeed. I also removed the `_packaged.html` files.

For publishing, I'd recommend the NPM package [gh-pages](https://www.npmjs.com/package/gh-pages#command-line-utility). Its usage is as simple as `npm run build && gh-pages build/`.

**Comment by tyrasd** [comment]
Great, thanks for the effort, Simon!

Is it correct that now everything is done except for the publishing step (which I'll take a look at asap – and thanks for the tip with the *gh-pages* package)?

**Comment by simon04** [comment]
Now, copying the `icons/` to `build/` is also handled by webpack. I _think_, all tasks are done. But please also try building the application and inspect the output in `build/`.

For running `gh-pages build/`, please be aware that this command not only makes a commit, but immediately pushes the result.

**Comment by mmd-osm** [comment]
Somehow this change seems to have broken the shortlink feature: the share button only produces lengthy URLs rather than the /s/xyz redirect.

**Comment by tyrasd** [comment]
@mmd-osm thanks for spotting that! I think I probably just forgot to set the respective setting when re-deploying on overpass-turbo.eu.

**Comment by tyrasd** [comment]
it should work again now. let me know if you find any more regressions. :+1:

**Comment by mmd-osm** [comment]
I'm not sure about this one. There's a "/missing translation/" shown instead of some German text in Autorun mode: http://overpass-turbo.eu/s/oSq

grafik

**Comment by tyrasd** [comment]
:+1: It's a regression. And I see this also in the English UI.

**Comment by mmd-osm** [comment]
Here's another one...

```C
[bbox:{{bbox}}];
node[place=village];
out;

{{style:
node{
  text: eval('tag("name")');
}
}}
```

Fails with: Error while evaluating mapcss evals TypeError: o.a.evalparser is undefined

**Comment by mmd-osm** [comment]
Hmmm.. somehow, a recent deployment broke the shortlink feature again.

**Comment by tyrasd** [comment]
oops. :see_no_evil: It should now work again.

**Comment by mmd-osm** [comment]
There's still something wrong with the new deployment: overpass-api.de reports as "(beta)", and all of my saved queries are gone in the meantime. :(

**Comment by tyrasd** [comment]
sorry again, I really have to fix up my deploy scripts :unamused:

It should work again now.

> even the overpass-ide_code in localStorage

Are you sure? I don't think that's something that could have happened only by me messing up the build parameters here. For me, my saved queries are accessible again now.

**Comment by mmd-osm** [comment]
Great, thanks a lot for the quick fix!

> Are you sure? I don't think that's something that could have happened only by me messing up the build parameters here. For me, my saved queries are accessible again now.

Yes, you're right. I only looked at the Firefox dev tools, which seems to limit the variable length. All good in turbo now.

**Pull request merged**